### PR TITLE
Package restructure

### PR DESCRIPTION
**Pull request by BenIVRE**
Complete package restructure to remove duplication and support UPM deployment

### CODE REVIEW

**Review by mike-ivre — APPROVED** [review]

**Review by juanma-ivre — APPROVED** [review]

File: Assets/RTLTMPro/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,9 @@ These are the release notes for the RTLTMPro UPM package.
 The github repository is: https://github.com/engage-xr/RTLTMPro
 The parent repository is: https://github.com/hk1ll3r/RTLTMPro
 
+# v3.4.7 (2024/08/14)
```
Comment (daniel-ivre) [inline Assets/RTLTMPro/CHANGELOG.md]: Is it a good idea to modify the base repo versioning? I think we should probably keep it and just use our patch tags as usual. If they release a new version we'll have problems.

File: Assets/RTLTMPro/Runtime/RTLTextMeshPro.cs [hunk 2]
```diff
@@ -103,7 +103,7 @@ public void UpdateText()
             if (originalText == null)
                 originalText = "";
 
-            if (!ForceFix)
+            if (!ForceFix && !TextUtils.IsRTLInput(originalText))
```
Comment (daniel-ivre) [inline Assets/RTLTMPro/Runtime/RTLTextMeshPro.cs]: We shouldn't do this

File: Assets/RTLTMPro/package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "com.engage.rtltmpro",
   "displayName": "RTL Text Mesh Pro",
-  "version": "3.4.6-engage.2",
+  "version": "3.4.7-engage",
```
Comment (daniel-ivre) [inline Assets/RTLTMPro/package.json]: This should be `3.4.6-engage.3` in my opinion

**Review by daniel-ivre — CHANGES_REQUESTED** [review]

File: Assets/RTLTMPro/Runtime/RTLTextMeshPro.cs (hunk 2 above)
Comment (matthew-ivre) [reply Assets/RTLTMPro/Runtime/RTLTextMeshPro.cs]: So this is back to setting LTR/RTL based on the first letter in the string, but only if the application is set to RTL?

I'm not sure what Dani's concern is, but switching on the first character seemed to produce fewer issues that we have now. It also seems like the simplest fix for the browser address bar - which could be a URL or an Arabic search term.

If we keep ForceFix as based on app language, I think we'll have an issue where if an Arabic user enters the following:
[English word 2] [Arabic word 2] [English word 1] [Arabic word 1]
It'll appear in a non-Arabic app as:
[Arabic word 1] [English word 1] [Arabic word 2] [English word 2] 

I'm not sure what the best approach is.

Comment (BenIVRE) [reply Assets/RTLTMPro/Runtime/RTLTextMeshPro.cs]: I didn't mean to add this to this PR, but that is setting it back to the way it was before.

File: Assets/RTLTMPro/package.json (hunk 3 above)
Comment (BenIVRE) [reply Assets/RTLTMPro/package.json]: Yes, I need to put it back. UMP supports semantic versioning, but I didn't see our -engage.2 showing up when it was set that way before.

File: Assets/RTLTMPro/CHANGELOG.md (hunk 1 above)
Comment (BenIVRE) [reply Assets/RTLTMPro/CHANGELOG.md]: I doubt we'll see a new version... They haven't made a release in 2 years.

**Review by matthew-ivre — APPROVED** [review]

**Review by daniel-ivre — APPROVED** [review]

**Review by daniel-ivre — APPROVED** [review]

**Pull request merged**